lib/logstash/inputs/beats.rb

```diff
@@ -275,20 +259,39 @@ def target_codec_on_field?
     !@target_codec_on_field.empty?
   end
 
-  def client_authentification?
+  def client_authentication_enabled?
+    if original_params.include?('ssl_client_authentication')
+      return client_authentication_optional? || client_authentication_required?
+    end
+
+    # Keep backward compatibility with the deprecated `ssl_verify_mode` until it's not removed.
+    # When it's explicitly set (or both settings are absent), it should use the ssl_certificate_authorities
+    # to enable/disable the client authentication. (even if ssl_verify_mode => none)
+    certificate_authorities_configured?
+  end
+
+  def certificate_authorities_configured?
     @ssl_certificate_authorities && @ssl_certificate_authorities.size > 0
   end
 
   def client_authentication_metadata?
-    @ssl_peer_metadata && ssl_configured? && client_authentification?
+    @ssl_enabled && @ssl_peer_metadata && ssl_configured? && client_authentication_enabled?
```

Comment:
I have added the `@ssl_enabled` here to fix an existing issue. When `ssl => false`, ` ssl_peer_metadata => true`, and `ssl_certificate_authorities` is configured, it tries to [extract the metadata ](https://github.com/logstash-plugins/logstash-input-beats/blob/main/lib/logstash/inputs/beats/message_listener.rb#L135) from the netty context and crashes as SSL is actually disabled.